Mathlib/LinearAlgebra/TensorProduct.lean

```diff
@@ -424,6 +424,18 @@ theorem map₂_mk_top_top_eq_top : Submodule.map₂ (mk R M N) ⊤ ⊤ = ⊤ :=
   exact Submodule.span_mono fun _ ⟨m, n, h⟩ => ⟨m, n, trivial, trivial, h⟩
 #align tensor_product.map₂_mk_top_top_eq_top TensorProduct.map₂_mk_top_top_eq_top
 
+theorem pure_if_absorbing_add (x : TensorProduct R M N)
```

Comment:
It's also hard to think of a circumstance when we would want to use this lemma (I note that you don't use it here). I'm not against adding it but it would be helpful if you could motivate why you want it.

Reply:
Right, so I opened a [thread](https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/What.20to.20do.20with.20small.2C.20self-contained.20lemmas.3F) about this on github a long time ago, and the indication seemed to be that it wouldn't hurt to add lemmas like this to mathlib.

The point is to include a useful criterion of when every element of a tensor product is pure, which is true in the case of $`S^{-1}R\otimes_R M`$ where we are working. This lemma was used in an earlier version of my theorem, but ended up not being useful later (when rephrasing things as `IsBaseChange` instead of `IsTensorProduct`). I decided to keep the theorem in hopes it would be useful to others, but am happy to remove it (since it ended up basically being trivial after a lot of golfing).